pkg/tool/aws-cli/aws-cli.go

```diff
@@ -0,0 +1,175 @@
+
+	//Download the latest awscli
+	err = os.MkdirAll(versionedDir, os.FileMode(0755))
+	if err != nil {
+		return fmt.Errorf("failed to create version-specific directory '%s': %w", versionedDir, err)
+	}
+
+	err = aws.DownloadAWSCLIRelease(url, fileExtension, versionedDir)
+	if err != nil {
+		return fmt.Errorf("failed to download aws cli: %w", err)
+	}
+
+	//Unzip binary Bundle
+	bundle := "aws-cli" + fileExtension
+	awsArchiveFilepath := filepath.Join(versionedDir, bundle)
+	awsNewInstallDir := filepath.Join(versionedDir, "aws-cli")
+
+	if fileExtension == ".zip" {
+		err = utils.Unzip(awsArchiveFilepath, versionedDir)
+		if err != nil {
+			return fmt.Errorf("failed to unarchive the aws-cli file '%s': %w", awsArchiveFilepath, err)
+		}
+		awsOldInstallDir = filepath.Join(versionedDir, "aws")
+		awsBinaryFilepath = filepath.Join(awsNewInstallDir, awsExec)
+		//Rename unzipped directory
+		err = os.Rename(awsOldInstallDir, awsNewInstallDir)
+		if err != nil {
+			return fmt.Errorf("error renaming directory %w", err)
+
+		}
+
+	} else {
+		cmd := exec.Command("pkgutil", "--expand-full", awsArchiveFilepath, awsNewInstallDir)
+		err = cmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to extract the aws-cli file '%s': %w", awsArchiveFilepath, err)
+		}
+		awsBinaryFilepath = filepath.Join(awsNewInstallDir, awsExec)
+	}
+
+	// Link as latest
+	latestFilePath := t.symlinkPath(latestDir)
+	err = os.Remove(latestFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to remove existing 'aws' binary at '%s': %w", latestDir, err)
+	}
+
+	err = t.createWrapper(latestFilePath, awsBinaryFilepath)
+	if err != nil {
+		return fmt.Errorf("failed to create aws cli squid proxy wrapper: %w", err)
+	}
+
+	return nil
+}
+
+// toolDir returns this tool's specific directory given the root directory all tools are installed in
+func (t *Tool) toolDir(rootDir string) string {
+	return filepath.Join(rootDir, "aws")
+}
+
+func (t *Tool) symlinkPath(latestDir string) string {
+	return filepath.Join(latestDir, "aws")
+}
+
+func (t *Tool) Remove(rootDir, latestDir string) error {
+	// Remove all binaries owned by this tool
+	toolDir := t.toolDir(rootDir)
+	err := os.RemoveAll(toolDir)
+	if err != nil {
+		return fmt.Errorf("failed to remove %s: %w", toolDir, err)
+	}
+
+	// Remove all symlinks owned by this tool
+	latestFilePath := t.symlinkPath(latestDir)
+	err = os.Remove(latestFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to remove symlinked file %s: %w", latestFilePath, err)
+	}
+	return nil
+}
+
+func (t *Tool) createWrapper(latestDir, awsPath string) error {
+	var builder strings.Builder
+	builder.WriteString(`#!/usr/bin/env bash
+set \
+  -o nounset \
+  -o pipefail \
+  -o errexit
+export HTTPS_PROXY=squid.corp.redhat.com:3128
+export HTTP_PROXY=squid.corp.redhat.com:3128
+`)
+	builder.WriteString(fmt.Sprintf("exec %s \"$@\"\n", awsPath))
+
+	input := builder.String()
+
+	err := os.WriteFile(latestDir, []byte(input), 0755)
```

Comment:
I think we should be generating this file in the `versionedDir` and symlinking it to `latest/`. This way, changes to the wrapper are also cached between versions